### ISSUE TITLE
[web:a11y] Enabling Semantics layer breaks DropdownMenu functionality in 3.19.0

### ISSUE THREAD

**Issue by fideldonson** [flutter/flutter#143848]
### Steps to reproduce

run SemanticsBinding.instance.ensureSemantics();
Implement basic DropdownMenu widget
Click on the dropdown arrow icon

### Expected results

Dropdown opens

### Actual results

Last DropdownMenuEntry is selected

### Code sample

<details open><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';
import 'package:flutter/rendering.dart';

void main() {
  WidgetsFlutterBinding.ensureInitialized();
  SemanticsBinding.instance.ensureSemantics();

  runApp(const MainApp());
}

class MainApp extends StatelessWidget {
  const MainApp({super.key});

  @override
  Widget build(BuildContext context) {
    return const MaterialApp(
      home: Scaffold(
        body: Center(
          child: DropdownMenu<int>(
            dropdownMenuEntries: [
              DropdownMenuEntry(value: 0, label: "zero"),
              DropdownMenuEntry(value: 1, label: "one"),
              DropdownMenuEntry(value: 2, label: "two"),
            ],
          ),
        ),
      ),
    );
  }
}

```

</details>

### Screenshots or Video

<details open>
<summary>Screenshots / Video demonstration</summary>

[Upload media here]

</details>

### Logs

<details open><summary>Logs</summary>

```console
[Paste your logs here]
```

</details>

### Flutter Doctor output

<details open><summary>Doctor output</summary>

```console
[Paste your output here]
```

</details>

**Comment by yjbanov** [comment]
When semantics is enabled the focus should go to the first entry. It's unclear why it goes to the last one. Will need to investigate.

**Comment by TalebRafiepour** [comment]
same issue

**Comment by rkunboxed** [comment]
This looks to be resolved in 3.24.1. I can no longer reproduce.

**Comment by yjbanov** [comment]
I need help from the framework team to understand the semantics tree generated by the framework.

Here's the semantics tree when dropdown menu is collapsed:

```
SemanticsNode#0
 │ Rect.fromLTRB(0.0, 0.0, 575.0, 1158.0)
 │
 └─SemanticsNode#1
   │ Rect.fromLTRB(0.0, 0.0, 575.0, 1158.0)
   │ textDirection: ltr
   │
   └─SemanticsNode#2
     │ Rect.fromLTRB(0.0, 0.0, 575.0, 1158.0)
     │ sortKey: OrdinalSortKey#b48f1(order: 0.0)
     │
     └─SemanticsNode#3
       │ Rect.fromLTRB(0.0, 0.0, 575.0, 1158.0)
       │ flags: scopesRoute
       │
       └─SemanticsNode#4
         │ Rect.fromLTRB(227.5, 555.0, 347.5, 603.0)
         │ actions: didGainAccessibilityFocus, didLoseAccessibilityFocus,
         │   focus, moveCursorBackwardByCharacter, moveCursorBackwardByWord,
         │   tap
         │ flags: isTextField, hasEnabledState, isEnabled
         │ value: "two"
         │ textDirection: ltr
         │ text selection: [3, 3]
         │ currentValueLength: 3
         │
         ├─SemanticsNode#10
         │ │ merge boundary ⛔️
         │ │ Rect.fromLTRB(0.0, 0.0, 120.0, 40.0)
         │ │
         │ └─SemanticsNode#11
         │     merged up ⬆️
         │     Rect.fromLTRB(0.0, 0.0, 120.0, 40.0)
         │     actions: tap
         │     flags: hasEnabledState, isEnabled
         │     label: "two"
         │     textDirection: ltr
         │
         ├─SemanticsNode#8
         │ │ merge boundary ⛔️
         │ │ Rect.fromLTRB(0.0, 0.0, 120.0, 40.0)
         │ │
         │ └─SemanticsNode#9
         │     merged up ⬆️
         │     Rect.fromLTRB(0.0, 0.0, 120.0, 40.0)
         │     actions: focus, tap
         │     flags: hasEnabledState, isEnabled, isFocusable
         │     label: "one"
         │     textDirection: ltr
         │
         ├─SemanticsNode#6
         │ │ merge boundary ⛔️
         │ │ Rect.fromLTRB(0.0, 0.0, 120.0, 40.0)
         │ │
         │ └─SemanticsNode#7
         │     merged up ⬆️
         │     Rect.fromLTRB(0.0, 0.0, 120.0, 40.0)
         │     actions: focus, tap
         │     flags: hasEnabledState, isEnabled, isFocusable
         │     label: "zero"
         │     textDirection: ltr
         │
         ├─SemanticsNode#12
         │   Rect.fromLTRB(4.0, 4.0, 44.0, 44.0)
         │   actions: focus, tap
         │   flags: isButton, hasEnabledState, isEnabled, isFocusable
         │
         └─SemanticsNode#5
             Rect.fromLTRB(76.0, 4.0, 116.0, 44.0)
             actions: focus, tap
             flags: isButton, hasEnabledState, isEnabled, isFocusable
```

Node 3 is the root of the screen (default route, I guess). Node 4 seems to correspond to the `DropdownMenu` widget. The internal text field seems to have been merged into it. A couple of concerns:

* Why are there nodes 10, 8, and 6, corresponding to buttons "two", "one", and "zero" in the semantics tree when the menu is not there? What is the framework trying to communicate to the engine? They seem to be just stacked on top of each other, and are overlaid by nodes 12 and 5, so it's unclear how to interact with them.
* These buttons are _children_ of the text field. I know how a text field works when it's a leaf node, but I'm not sure how to interpret a text field that's a container of other nodes. How should the engine interpret such structure?

When the menu is shown the semantics tree looks like this:

```
SemanticsNode#0
 │ Rect.fromLTRB(0.0, 0.0, 575.0, 1158.0)
 │
 └─SemanticsNode#1
   │ Rect.fromLTRB(0.0, 0.0, 575.0, 1158.0)
   │ textDirection: ltr
   │
   └─SemanticsNode#2
     │ Rect.fromLTRB(0.0, 0.0, 575.0, 1158.0)
     │ sortKey: OrdinalSortKey#b48f1(order: 0.0)
     │
     └─SemanticsNode#3
       │ Rect.fromLTRB(0.0, 0.0, 575.0, 1158.0)
       │ flags: scopesRoute
       │
       ├─SemanticsNode#4
       │ │ Rect.fromLTRB(227.5, 555.0, 347.5, 603.0)
       │ │ actions: didGainAccessibilityFocus, didLoseAccessibilityFocus,
       │ │   focus, moveCursorBackwardByCharacter, moveCursorBackwardByWord,
       │ │   tap
       │ │ flags: isTextField, hasEnabledState, isEnabled
       │ │ value: "two"
       │ │ textDirection: ltr
       │ │ text selection: [3, 3]
       │ │ currentValueLength: 3
       │ │
       │ ├─SemanticsNode#10
       │ │ │ merge boundary ⛔️
       │ │ │ Rect.fromLTRB(0.0, 0.0, 120.0, 40.0)
       │ │ │
       │ │ └─SemanticsNode#11
       │ │     merged up ⬆️
       │ │     Rect.fromLTRB(0.0, 0.0, 120.0, 40.0)
       │ │     actions: tap
       │ │     flags: hasEnabledState, isEnabled
       │ │     label: "two"
       │ │     textDirection: ltr
       │ │
       │ ├─SemanticsNode#8
       │ │ │ merge boundary ⛔️
       │ │ │ Rect.fromLTRB(0.0, 0.0, 120.0, 40.0)
       │ │ │
       │ │ └─SemanticsNode#9
       │ │     merged up ⬆️
       │ │     Rect.fromLTRB(0.0, 0.0, 120.0, 40.0)
       │ │     actions: focus, tap
       │ │     flags: hasEnabledState, isEnabled, isFocusable
       │ │     label: "one"
       │ │     textDirection: ltr
       │ │
       │ ├─SemanticsNode#6
       │ │ │ merge boundary ⛔️
       │ │ │ Rect.fromLTRB(0.0, 0.0, 120.0, 40.0)
       │ │ │
       │ │ └─SemanticsNode#7
       │ │     merged up ⬆️
       │ │     Rect.fromLTRB(0.0, 0.0, 120.0, 40.0)
       │ │     actions: focus, tap
       │ │     flags: hasEnabledState, isEnabled, isFocusable
       │ │     label: "zero"
       │ │     textDirection: ltr
       │ │
       │ ├─SemanticsNode#12
       │ │   Rect.fromLTRB(4.0, 4.0, 44.0, 44.0)
       │ │   actions: focus, tap
       │ │   flags: isSelected, isButton, isFocused, hasEnabledState,
       │ │     isEnabled, isFocusable
       │ │
       │ └─SemanticsNode#5
       │     Rect.fromLTRB(76.0, 4.0, 116.0, 44.0)
       │     actions: focus, tap
       │     flags: isSelected, isButton, hasEnabledState, isEnabled,
       │       isFocusable
       │
       └─SemanticsNode#101
         │ Rect.fromLTRB(227.5, 603.0, 347.5, 763.0)
         │ thickness: 3.0
         │
         └─SemanticsNode#102
           │ Rect.fromLTRB(0.0, 8.0, 120.0, 152.0)
           │ flags: hasImplicitScrolling
           │ scrollExtentMin: 0.0
           │ scrollPosition: 0.0
           │ scrollExtentMax: 0.0
           │ elevation: 3.0
           │
           ├─SemanticsNode#103
           │ │ merge boundary ⛔️
           │ │ Rect.fromLTRB(0.0, 0.0, 120.0, 48.0)
           │ │
           │ └─SemanticsNode#104
           │     merged up ⬆️
           │     Rect.fromLTRB(0.0, 0.0, 120.0, 48.0)
           │     actions: focus, tap
           │     flags: hasEnabledState, isEnabled, isFocusable
           │     label: "zero"
           │     textDirection: ltr
           │
           ├─SemanticsNode#105
           │ │ merge boundary ⛔️
           │ │ Rect.fromLTRB(0.0, 48.0, 120.0, 96.0)
           │ │
           │ └─SemanticsNode#106
           │     merged up ⬆️
           │     Rect.fromLTRB(0.0, 0.0, 120.0, 48.0)
           │     actions: focus, tap
           │     flags: hasEnabledState, isEnabled, isFocusable
           │     label: "one"
           │     textDirection: ltr
           │
           └─SemanticsNode#107
             │ merge boundary ⛔️
             │ Rect.fromLTRB(0.0, 96.0, 120.0, 144.0)
             │
             └─SemanticsNode#108
                 merged up ⬆️
                 Rect.fromLTRB(0.0, 0.0, 120.0, 48.0)
                 actions: focus, tap
                 flags: hasEnabledState, isEnabled, isFocusable
                 label: "two"
                 textDirection: ltr
```

Note that nodes 10, 8, and 6 are still there, and are still stacked on top of each other. However, there are now additional copies of the same buttons, represented by nodes 104, 105, and 107. What role do the copies play?

**Comment by chunhtai** [comment]
yeah, this doesn't look right. I will have to look into this

**Comment by hannah-hyj** [comment]
this should be fixed by https://github.com/flutter/flutter/pull/156709
